src/Microsoft.OData.Core/UriParser/Aggregation/AggregationMethod.cs

```diff
@@ -34,6 +34,12 @@ public enum AggregationMethod
         /// <summary>
         /// The aggregation method CountDistinct.
         /// </summary>
-        CountDistinct
+        CountDistinct,
+
+        /// <summary>
+        /// The aggregation method Count.
+        /// Used only internally to represent the virtual property $count.
+        /// </summary>
+        VirtualPropertyCount,

```

Comment:
The doc says it's a virtual property. If I'm not mistaken, ODL does not have any other virtual property implemented. Since I didn't find any guideline to follow about implementing these properties, I created a internal aggregation method count and mapped each call of the virtual property to it. This internal method is not visible to the end user, unlike the other aggregation methods.
I'd be happy to know about better approaches to implement it.